amo2kinto/importer.py

```diff
@@ -28,6 +29,11 @@ def sync_records(amo_records, fields,
                  kinto_client, bucket, collection, schema, permissions):
 
     amo_records = prepare_amo_records(amo_records, fields)
+
+    if schema:
+        for record in amo_records:
+            jsonschema.validate(record, schema)
```

Comment:
> To answer this question we should answer: Why do we want local validation?

Yes, it is a very useful safety check. As you said, either to check the existing records against a new schema, or check that the new ones are compatibles with the schema in place.